src/components/RightPane/CoursePane/SearchForm/CourseNumberSearchBar.tsx

```diff
@@ -7,14 +7,44 @@ interface CourseNumberSearchBarState {
     courseNumber: string;
 }
 
-class CourseNumberSearchBar extends PureComponent<Record<string, never>, CourseNumberSearchBarState> {
+
+class CourseNumberSearchBar extends PureComponent<Record<string,never>, CourseNumberSearchBarState> {
+
+
     state = {
-        courseNumber: RightPaneStore.getFormData().courseNumber,
+        courseNumber: RightPaneStore.getUrlCourseNumValue() != "null" && RightPaneStore.getUrlCourseNumValue() != "" && RightPaneStore.getUrlCourseNumValue() != " " 
+        ?
+        this.updateCourseNumAndGetFormData()
+        : 
+        RightPaneStore.getFormData().courseNumber,
+    };
+    
+    updateCourseNumAndGetFormData(){
+        RightPaneStore.updateFormValue("courseNumber", RightPaneStore.getUrlCourseNumValue()) 
+        return RightPaneStore.getFormData().courseNumber
```

Comment:
I don't like this, but I can see why this might be necessary if the store lives completely detached from component lifecycles.

Please consolidate this into a single method; e.g.
```ts
state = {
   courseNumber: this.getCourseNumber()
}
```

The `getCourseNumber` method will be responsible for syncing both the state of the store and the component; it will abstract away the ternaries, and nested function calls.